Source/WebCore/bindings/js/JSDOMWindowBase.cpp

```diff
@@ -258,16 +301,13 @@ void JSDOMWindowBase::queueMicrotaskToEventLoop(JSGlobalObject& object, Ref<JSC:
     auto userGestureToken = UserGestureIndicator::currentUserGesture();
     if (userGestureToken && (!userGestureToken->isPropagatedFromFetch() || !objectScriptExecutionContext->settingsValues().userGesturePromisePropagationEnabled))
         userGestureToken = nullptr;
-    eventLoop.queueMicrotask([task = WTFMove(task), userGestureToken = WTFMove(userGestureToken)]() mutable {
-        auto* globalObject = task->globalObject();
-        if (!userGestureToken) {
-            JSExecState::runTask(globalObject, task.get());
-            return;
-        }
-
-        UserGestureIndicator gestureIndicator(userGestureToken, UserGestureToken::GestureScope::MediaOnly, UserGestureToken::IsPropagatedFromFetch::Yes);
-        JSExecState::runTask(globalObject, task.get());
-    });
+
+    if (!userGestureToken)
+        task.setDispatcher(eventLoop.jsMicrotaskDispatcher());
+    else
+        task.setDispatcher(UserGestureInitiatedMicrotaskDispatcher::create(eventLoop, Ref { *userGestureToken }));
```

Comment:
Hm... it seems a bit wasteful to create a new object for every user gesture initiated task but it's no worse than the old code.